### DISCUSSION TITLE
Single Item Per User

### DISCUSSION THREAD

**Discussion by jsthind** [directus/directus#3155] [Feature Requests]
Is it possible to create a single item collection but with one item per user? That is, each user can edit their own item (row entry).

**Comment by rijkvanzanten** [comment]
Cool idea! It's not possible currently, but I can definitely see the use for this 👍

**Comment by danieljwestman** [comment]
Isn’t this kind of doable with custom permission?

**Reply by rijkvanzanten** [reply]
I suppose you're right actually 🤔 By adding an owner field to the singleton collection it might already work!

I've never tested this though, but in theory you're right

**Reply by danieljwestman** [reply]
I’m pretty sure I’ve tried this before, pretty straight forward (and flexible) from what I remember

**Reply by jsthind** [reply]
Singleton with owner field does not work in Directus v8. Haven't tried it out with the Directus v9 though, but v9 looks a little rough around the edge at the moment.

**Reply by benhaynes** [reply]
Could you describe why it looks "rough around the edges"? V9 is far more stable and polished than v8.

**Reply by jsthind** [reply]
It is mainly documentation issues. Most of the documentation still seems to be about the older version. For example, the "Storage Adapters" page still talks about "_composer require league/flysystem-aws-s3-v3_" but the v9 does not use PHP (at least that is what it looks to me).

For another example, see the screenshot at https://pasteboard.co/JBIAcsl.png

I can't figure out how to set custom permissions. In v8 we could select from the dropdown (Mine Only/Role Only). 

So it is mainly documentation issues. I don't want to sound ungrateful, Directus is an amazing app. It is just that I found about Directus recently (a month or so back) and the shift from PHP to node.js has been disconcerting.

**Reply by benhaynes** [reply]
You're probably just looking at the wrong Docs... the v9 Docs will be pushed this week, and they are staged here:

https://next.docs.directus.io/

These docs are far from complete, and most links are not set yet (since the pages are in flux), but our plan is to have the completed before 9.0.0 is released. 

Lastly, I'm not sure why adding the PHP port is disconcerting. It's good to keep things moving forward, and we didn't abandon PHP. We have v8 in PHP and v9 being rebuilt in Laravel as we speak... it's just moving slower than Node.

**Reply by jsthind** [reply]
Thank you, the updated docs will definitely help a lot.

So I tried out the following in v9:

1. Set the collection as a singleton gave full permissions to the related role. All users in this role see the same data.  If one user updates the data, it updates for everyone.
2. Set the collection as a singleton, and then gave custom permissions to the related role. Now only one user can edit the data, every other user gets a 403 error.

I used the following permission for read/edit/delete:

`{
    "user_created": {
        "_eq": "$CURRENT_USER"
    }
}`

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
